LayoutTest/TestCase/LYTLayoutFailingTestSnapshotRecorder.m

```diff
@@ -0,0 +1,233 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+#import "LYTLayoutFailingTestSnapshotRecorder.h"
+#import "LYTConfig.h"
+#import "LYTLayoutTestCase.h"
+
+void SimpleLog(NSString *format, ...) {
+    // Log to the console without timestamp that NSLog gives us
+    va_list args;
+    va_start(args, format);
+    NSString *formattedString = [[NSString alloc] initWithFormat:format
+                                                       arguments:args];
+    va_end(args);
+    [[NSFileHandle fileHandleWithStandardOutput] writeData:[formattedString dataUsingEncoding:NSNEXTSTEPStringEncoding]];
+}
+
+@interface LYTLayoutFailingTestSnapshotRecorder ()
+
+@property (nonatomic) Class invocationClass;
+@property (nonatomic) NSMutableSet *failingTestsSnapshotFolders;
+
+@end
+
+@implementation LYTLayoutFailingTestSnapshotRecorder
+
++ (instancetype)sharedInstance {
+    static LYTLayoutFailingTestSnapshotRecorder *sharedInstance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[LYTLayoutFailingTestSnapshotRecorder alloc] init];
+    });
+    return sharedInstance;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.failingTestsSnapshotFolders = [NSMutableSet set];
+        XCTestObservationCenter *observationCenter = [XCTestObservationCenter sharedTestObservationCenter];
+        [observationCenter addTestObserver:self];
+    }
+    return self;
+}
+
+- (void)testBundleDidFinish:(NSBundle *)testBundle {
+    if (self.failingTestsSnapshotFolders.count > 0) {
+        SimpleLog(@"\nSnapshots of failing tests can be found in:\n");
+        for (NSString *path in self.failingTestsSnapshotFolders) {
+            SimpleLog(@"%@\n", path);
+        }
+        [self.failingTestsSnapshotFolders removeAllObjects];
+    }
+}
+
+- (void)testCase:(XCTestCase *)testCase didFailWithDescription:(NSString *)description inFile:(nullable NSString *)filePath atLine:(NSUInteger)lineNumber {
+    if ([testCase isKindOfClass:[LYTLayoutTestCase class]]) {
+        NSString *pathForSnapshots = [self commonRootPathForInvocationClass:[testCase class]];
+        pathForSnapshots = [pathForSnapshots stringByAppendingString:@"/index.html"];
+        [self.failingTestsSnapshotFolders addObject:pathForSnapshots];
+    }
+}
+
+- (void)startNewLogForClass:(Class)invocationClass {
+    self.invocationClass = invocationClass;
+    [self deleteCurrentFailingSnapshots];
```

Comment:
This confused me for a while. It looks like you're deleting all the snapshots...but you're just deleting the ones for the new invocation. Same with the one below.

Thoughts on making this more functional?
as in: deleteCurrentFailingSnapshotsForInvocationClass:

This way things like the order of lines here don't matter. Obviously this works too though, just wondered what you think. (same for methods below).